lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/GroupCommDriver.cs

```diff
@@ -147,6 +148,24 @@ public ICommunicationGroupDriver NewCommunicationGroup(string groupName, int num
         }
 
         /// <summary>
+        /// Remove a group from the GroupCommDriver
+        /// Throw ArgumentException if the group does not exist
+        /// </summary>
+        /// <param name="groupName"></param>
+        public void RemoveCommunicationGroup(string groupName)
+        {
+            lock (_groupsLock)
```

Comment:
Usage of `_groupsLock` is not consistent in this file, e.g. `line 118`. Would also advise to put `_groupsLock` around methods like `NewCommunicationGroup`.